### PR TITLE
Post Uploading Criteria Changed Completely

### PR DESCRIPTION
**Pull request by Gurmeet-Singh657**
This is how creating new posts look like now
image

This is how when new post with caption & Image uploaded
image

@narayan954 Plz merge my pull request !!

### CODE REVIEW

**Comment by narayan954** [comment]
> This is how creating new posts look like now image
> 
> This is how when new post with caption & Image uploaded image
> 
> @narayan954 Plz merge my pull request !!

please mention issue number

**Comment by Gurmeet-Singh657** [comment]
Issue number 383 😅. Branch name was issue number itself !!

**Comment by narayan954** [comment]
> Issue number 383 sweat_smile. Branch name was issue number itself !!

oh, my bad but still mention it in the pr as well

**Review by narayan954 — CHANGES_REQUESTED** [review]
When clicking on add new post button, it tried to load the imgupload component but results in crash due to loss of access to user prop... Can you try to fix this??
image
I am not able to check your feature as it's crashing.

**Comment by narayan954** [comment]
If you can move this to the sidebar's new post button and implement it for there, it'd be really helpful. I can change the issue difficulty level to 3 as it might be tedious, removing code from there and placing there and ensuring the working

**Comment by Gurmeet-Singh657** [comment]
Can you plz give an Example that where to shift this ?

**Comment by Gurmeet-Singh657** [comment]
@narayan954 OK Done I will change it ✅.

**Comment by narayan954** [comment]
> Can you plz give an Example that where to shift this ?

check the current deployed version of our application, we've a side bar with new post button  as well which is un implemented. If you wish to fix that, you can... Else it's okay we'll handle that in a different issue.You can fix the current issue I highlighted.

**Comment by Gurmeet-Singh657** [comment]
I will do it , I have no problem. I have to remove New post button from above Navbar and that new post dialog should be shown on clicking sidebar new post . Right ??

**Comment by narayan954** [comment]
> I will do it , I have no problem. I have to remove New post button from above Navbar and that new post dialog should be shown on clicking sidebar new post . Right ??

yes, exactly... It'll cause a bit of code refactoring as well. We're now focussing on code refactoring as well.

**Comment by Gurmeet-Singh657** [comment]
No Problem !! I have started working on this functionality !! Plz mark it as Level 3 issue as you said 🙂

**Review by narayan954 — APPROVED** [review]
Awesome job @Gurmeet-Singh657 this looks great and you did a lot of work on this! Thank you!

File: src/components/SideBar.jsx [hunk 1]
```diff
@@ -42,11 +48,58 @@ function SideBar(props) {
             })
           }
         >
-          <div className="profile">
+          <div className="sidebar_align">
             <AccountCircleIcon className="icon" /> <span>Profile</span>
           </div>
         </li>
       </ul>
+      <Dialog
```
Comment (narayan954) [inline src/components/SideBar.jsx:56]: later we may put this on a separate component NewPost. I'll be doing some refactoring as well.

Comment (Gurmeet-Singh657) [reply src/components/SideBar.jsx:56]: I want to do this work !! So, Should I create a new Issue !! 😅

**Pull request merged**